agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/HttpServerTracer.java

```diff
@@ -16,35 +16,14 @@
 
 package io.opentelemetry.auto.bootstrap.instrumentation.decorator;
 
-import static io.opentelemetry.OpenTelemetry.getPropagators;
-import static io.opentelemetry.context.ContextUtils.withScopedContext;
-import static io.opentelemetry.trace.Span.Kind.SERVER;
-import static io.opentelemetry.trace.TracingContextUtils.getSpan;
-import static io.opentelemetry.trace.TracingContextUtils.withSpan;
 
-import io.grpc.Context;
 import io.opentelemetry.OpenTelemetry;
-import io.opentelemetry.auto.config.Config;
-import io.opentelemetry.auto.instrumentation.api.MoreTags;
-import io.opentelemetry.auto.instrumentation.api.Tags;
-import io.opentelemetry.context.Scope;
-import io.opentelemetry.context.propagation.HttpTextFormat;
 import io.opentelemetry.trace.Span;
-import io.opentelemetry.trace.SpanContext;
 import io.opentelemetry.trace.Tracer;
-import io.opentelemetry.trace.attributes.SemanticAttributes;
-import java.io.PrintWriter;
-import java.io.StringWriter;
 import java.lang.reflect.Method;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.util.concurrent.ExecutionException;
-import lombok.extern.slf4j.Slf4j;
 
 // TODO In search for a better home package
-@Slf4j
-public abstract class HttpServerTracer<REQUEST> {
-  public static final String CONTEXT_ATTRIBUTE = "io.opentelemetry.instrumentation.context";
+public abstract class HttpServerTracer<REQUEST> extends HttpServerTracerBase<REQUEST> {
 
   protected final Tracer tracer;
 
```

Comment:
I moved most of logic from HttpServerTracer to HttpServerTracerBase. This new base class contains all logic from HttpServerTracer but it does not have a tracer field, instead a tracer object supplied to the `startSpan()` and `getCurrentSpan()` methods. I need this change to use my own tracer in the SpringWebMVCDecorator.java defined in instrumentation-core:spring:spring-webmvc-3.1-core. With this change all the classes that currently depend on HttpServerTracer will remain unaffected. This implementation limited the number of code changes I needed to make to use this decorator for manual instrumentation.